src/main/java/uk/gov/ch/oauth/IOauth.java

```diff
@@ -1,18 +1,47 @@
 package uk.gov.ch.oauth;
 
-import uk.gov.companieshouse.session.Session;
-
 import javax.servlet.http.HttpServletResponse;
+import uk.gov.companieshouse.session.Session;
 
+/**
+ * Handler for the OAuth sight in flow based on the requirements of the CH version of OAuth2
+ */
 public interface IOauth {
 
+    /**
+     * Only referenced in tests and an unused method in {@link uk.gov.ch.developer.docs.controller.developer.SignInController}
+     *
+     * @param attributeName always "email"
+     * @return ?
+     * @see uk.gov.ch.developer.docs.controller.developer.SignInController#createAuthoriseURIWithForceAndHint(String,
+     * String, String, String)
+     */
     String oauth2EncodeState(final String returnUri,
```

Comment:
Not strictly true and largely a problem of how we refactored perhaps?
This method previously was the main login one before encodeSignInState was  added to wrap this and allow a session to be passed.
This maybe should have been removed from the interface then?

Reply:
Unless I've gotten confused somewhere